CONTRIBUTING.md

```diff
@@ -1,34 +1,162 @@
 # Contributing
 
-## Pull Request Process
+## Changes to the CI Infrastructure
+
+Changes to the CI Infrastructure are made to the `main` branch in this repository. Config branches use the `call-*.yml` workflows to `workflow_call` the equivalent workflow that is on the `main` branch.
+
+Since the logic in the CI infrastructure is quite involved, it would be a good idea to read the [README-DEV.md](./README-DEV.md).
+
+## Dev and Release branches
+
+Each configuration has a `dev-*` and `release-*` branch. They differ in the CI checks that are run when pull requests are made to update the branch.
+
+### Release
 
-### Creation of a new ACCESS-OM2 Config
+Pull requests to the `release-*` branch are intended to create a new version of the configuration. It is expected that the version *will* be updated before the PR can be merged. This in turn creates a GitHub release. It can be confusing for users if there are a large number of versions of a configuration. For this reason the atomicity of updates to a released configuration should be minimised, i.e.  updates should be meaningful.
```

Comment:
```suggestion
Pull requests to the `release-*` branch are intended to create a new version of the configuration. It is expected that the version *will* be updated before the PR can be merged. This in turn creates a new tag for that configuration branch. It can be confusing for users if there are a large number of versions of a configuration. For this reason the atomicity of updates to a released configuration should be minimised, i.e.  updates should be meaningful.
```

Reply:
Just because a GitHub Release has a different meaning - https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases